src/splinefunctions.cpp

```diff
@@ -0,0 +1,779 @@
+                }
+                break;
+
+            default:
+                /* We don't need specific coefficients in the cases of:
+                 * noExtrapolation, polynomial, periodic*/
+                break;
+        }
+        /* Write them to the vector */
+        coefficients_extrapolate_sensi[4 * ip] = sp0 - sm0 * nodes_[0];
+        coefficients_extrapolate_sensi[4 * ip + 1] = sm0;
+    }
+
+    realtype sp_end;
+    realtype sm_end;
+    for (int ip = 0; ip < nplist; ip++) {
+        sp_end = dspline_valuesdp[spline_offset + ip + (n_nodes()-1) * nplist];
+        switch (lastNodeEP_) {
+            /* This whole switch-case-if-else-if-thing could be moved
+             * outside the loop, I know. Yet, it's at most some thousand
+             * if's done once in the program for saving many lines of code
+             * and getting a much clearer code structure. */
+            case SplineExtrapolation::constant:
+                sm_end = 0;
+                break;
+
+            case SplineExtrapolation::linear:
+                if (get_node_derivative_by_FD() &&
+                    lastNodeBC_ == SplineBoundaryCondition::given) {
+                    sm_end = (sp_end - dspline_valuesdp[spline_offset +
+                              ip + (n_nodes() - 2) * nplist])
+                             / (nodes_[n_nodes() - 1] - nodes_[n_nodes() - 2]);
+
+                } else if (get_node_derivative_by_FD() &&
+                           lastNodeBC_ == SplineBoundaryCondition::natural) {
+                    throw AmiException("Natural boundary condition for "
+                        "Hermite splines with linear extrapolation is "
+                        "not yet implemented.");
+
+                } else if (!get_node_derivative_by_FD() &&
+                           lastNodeBC_ == SplineBoundaryCondition::given) {
+                    sm_end = dspline_slopesdp[spline_offset + ip +
+                             (n_nodes() - 1) * nplist];
+
+                } else if (!get_node_derivative_by_FD() &&
+                           lastNodeBC_ == SplineBoundaryCondition::natural) {
+                    throw AmiException("Natural boundary condition for "
+                        "Hermite splines with linear extrapolation is "
+                        "not yet implemented.");
+
+                } else {
+                    throw AmiException("Some weird combination of spline boundary "
+                        "condition, extrapolation and finite differecnces was "
+                        "passed which should bo ne allowed.");
+                }
+                break;
+
+            default:
+                /* We don't need specific coefficients in the cases of:
+                 * noExtrapolation, polynomial, periodic*/
+                break;
+        }
+        /* Write them to the vector */
+        coefficients_extrapolate_sensi[4 * ip + 2] = sp_end -
+            sm_end * nodes_[n_nodes() - 1];
```

Comment:
@lcontento : If the `default` branch is taken, `sm_end` is undefined here. What is supposed to happen?

Reply:
Not 100% sure. I have some uncommitted changes to fix the compilation mistakes and there I just "fixed" it by raising an exception in the default case...